tools/Repo-Tasks.psm1

```diff
@@ -0,0 +1,253 @@
+        [ValidateNotNullOrEmpty()]
+        [string]$ServicePrincipal,
+
+        [parameter(Mandatory=$false, Position=4, HelpMessage='ServicePrincipal Secret/ClientId Secret you would like to use')]
+        [ValidateNotNullOrEmpty()]
+        [string]$ServicePrincipalSecret,
+
+        [parameter(Mandatory=$false, Position=5, HelpMessage='RecordMode you would like to set')]    
+        [string]$RecordMode = "Playback",
+
+        [parameter(Mandatory=$false, Position=6, HelpMessage='Target Environment to use {Prod (default) | Dogfood | Next | Current')]
+        [string]$TargetEnvironment = "Prod"
+    )
+
+    [string]$uris="BaseUri=https://management.azure.com/;AADAuthEndpoint=https://login.windows.net/;GraphUri=https://graph.windows.net/"
+
+    $formattedConnStr = [string]::Format("SubscriptionId={0};AADTenant={1};HttpRecorderMode={2};Environment={3}", $SubscriptionId, $TenantId, $RecordMode, $TargetEnvironment)
+
+    if([string]::IsNullOrEmpty($UserId) -eq $false)
+    {
+        $formattedConnStr = [string]::Format([string]::Concat($formattedConnStr, ";UserId={0}"), $UserId)
+    }
+
+    if([string]::IsNullOrEmpty($ServicePrincipal) -eq $false)
+    {
+        $formattedConnStr = [string]::Format([string]::Concat($formattedConnStr, ";ServicePrincipal={0}"), $ServicePrincipal)
+    }
+
+    if([string]::IsNullOrEmpty($ServicePrincipalSecret) -eq $false)
+    {
+        $formattedConnStr = [string]::Format([string]::Concat($formattedConnStr, ";ServicePrincipalSecret={0}"), $ServicePrincipalSecret)
+    }
+
+    $formattedConnStr = [string]::Concat($formattedConnStr, ";", $uris)
+
+    #Set connection string to Environment variable
+    $env:TEST_CSM_ORGID_AUTHENTICATION=$formattedConnStr
+    Write-Host ""
+
+    # Retrieve the environment variable
+    Write-Host "Setting up below connection string. Start Visual Studio by typing devenv" -ForegroundColor Green
+    [Environment]::GetEnvironmentVariable($envVariableName)
+    Write-Host ""
+    
+    Write-Host "If your needs demand you to set connection string differently, for all the supported Key/Value pairs in connection string"
+    Write-Host "Please visit https://github.com/Azure/azure-powershell/blob/dev/documentation/Using-Azure-TestFramework.md"
+}
+
+[CmdletBinding]
+Function Get-BuildScopes
+{
+<#
+.SYNOPSIS
+You can build a particular package rather than doing a full build by providing Build Scope.
+This cmdlet will help to identify existing Scope available
+This will enable to execute Start-RepoBuild <scope>
+
+#>
+
+    Write-Host "Below are available scopes you can specify for building specific projects"
+    Write-Host ""    
+    Get-ChildItem -path "$env:repoRoot\src\ResourceManager" -dir | Format-Wide -Column 5 | Format-Table -Property Name    
+    Get-ChildItem -path "$env:repoRoot\src\ServiceManagement" -dir | Format-Wide -Column 5 | Format-Table -Property Name
+}
+
+[CmdletBinding]
+Function Start-RepoBuild
```

Comment:
Other modules are likely to use Start-Build so I would suggest a rename to Start-AzureRMBuild